src/codeflare_sdk/cluster/cluster.py

```diff
@@ -652,56 +675,57 @@ def get_cluster(
-
-                            ingress_options = {
-                                "ingresses": [
-                                    {
-                                        "ingressName": ingress_name,
-                                        "port": port,
-                                        "annotations": annotations,
-                                        "ingressClassName": ingress_class_name,
-                                        "pathType": path_type,
-                                        "path": path,
-                                        "host": ingress_host,
-                                    }
-                                ]
-                            }
-            except Exception as e:
-                return _kube_api_error_handling(e)
+                            ingress_host = ingress.spec.rules[0].host
+                            if (
+                                "ingress-options" in ingress.metadata.labels
+                                and ingress.metadata.labels["ingress-options"] == "true"
+                            ):
+                                ingress_name = ingress.metadata.name
+                                port = (
+                                    ingress.spec.rules[0]
+                                    .http.paths[0]
+                                    .backend.service.port.number
+                                )
+                                annotations = ingress.metadata.annotations
+                                path = ingress.spec.rules[0].http.paths[0].path
+                                ingress_class_name = ingress.spec.ingress_class_name
+                                path_type = (
+                                    ingress.spec.rules[0].http.paths[0].path_type
+                                )
+
+                                ingress_options = {
+                                    "ingresses": [
+                                        {
+                                            "ingressName": ingress_name,
+                                            "port": port,
+                                            "annotations": annotations,
+                                            "ingressClassName": ingress_class_name,
+                                            "pathType": path_type,
+                                            "path": path,
+                                            "host": ingress_host,
+                                        }
+                                    ]
+                                }
+                except Exception as e:  # pragma: no cover
+                    return _kube_api_error_handling(e)
             # We gather the ingress domain from the host
             if ingress_host is not None and ingress_options == {}:
                 ingress_domain = ingress_host.split(".", 1)[1]
             else:
                 ingress_domain = None
```

Comment:
Not needed for this PR but I'm pretty sure this logic can be moved up into the conditional on L688 so that no extra conditionals are necessary since ingress_host and ingress_options are both set conditionally already.